src/ladder.ts

```diff
@@ -46,11 +46,11 @@ export class Ladder {
 		this.rpoffset = 9 * 60 * 60;
 	}
 	getRP() {
-		const rpnum = ((time() - this.rpoffset) / this.rplen) + 1;
+		const rpnum = Math.trunc((time() - this.rpoffset) / this.rplen) + 1;
 		return rpnum * this.rplen + this.rpoffset;
 	}
 	nextRP(rp: number) {
-		const rpnum = (rp / this.rplen);
+		const rpnum = Math.trunc(rp / this.rplen);
```

Comment:
These two changes clearly bring the TypeScript implement closer to [the PHP version](https://github.com/smogon/pokemon-showdown-client/blob/master/lib/ntbb-ladder.lib.php#L121-L128), which used `intval()` on these.